src/trainer.py

```diff
@@ -1016,23 +1022,26 @@ def _description_from_metrics(self, metrics):
         """ format some metrics as a string """
         return ", ".join(["%s: %.4f" % (name, value) for name, value in metrics.items()])
 
-    def _unmark_previous_best(self, phase, epoch):
+    def _unmark_previous_best(self, phase, epoch, task=""):
         marked_best = glob.glob(
-            os.path.join(self._serialization_dir, "*_state_{}_epoch_*.best_macro.th".format(phase))
+            os.path.join(
+                self._serialization_dir + task, "*_state_{}_epoch_*.best_macro.th".format(phase)
```

Comment:
There are still a few instances of best_macro floating around. We should only ever write ".best" as a filename, and if we do want to keep reading/searching for ".best_macro" as a backward compatibility measure (optional), we should also read ".best".

Reply:
I intentionally put in best_macro as the variable name because what we are doing is tracking macro averages (which if you have only one task, it is equal to tracking its performance). If there's a strong reason why you think .best > .best_macro, I can change it

Reply:
Yeah, we can get more votes from others if you want, but I think 'best' is a lot clearer. It's more standard, and most users likely won't be doing MTL pretraining.